### DISCUSSION TITLE
Help us make our Libraries work with j2cl

### DISCUSSION THREAD

**Discussion by jweiland1** [google/j2cl#163] [Q&A]
The[Virtual FTC simulator](https://www.vrobotsim.online/homepage.html) team is looking for someone (ideally a FIRST ALUMN)who wants a programming challenge. We want your help adapting j2cl to transpile Java code to JS. The Programmer will be working on automatic translation of Java to Javascript.
We tried to use JSweet  but hit a wall- Interested? jweiland@firstillinoisrobotics.org

here is our story with jsweet https://docs.google.com/document/d/1cqJP6u8igPTF3kmU5TarHcQ-vrfobMsDW0wDVTckMDM/edit?usp=sharing

**Comment by niloc132** [comment]
Consider taking another look at jsweet - the `409 Conflict`, `403 Forbidden` errors are just maven setup problems (most likely not listing the correct repository server for the required dependencies) - these setup problems won't be solved by moving to another tool.  Your writeup also doesn't seem to include any of the work done so far, perhaps to make suggestions on how to improve it (or understand the scope, since as discussed below, the majority of the code I saw doesn't make sense to run in the browser).

There is presently no easy out of the box integration to use j2cl in gradle, though bazel and maven are well supported. With that said, given the issues already encountered with maven, it is likely you'll encounter some of the same problems - and bazel is a learning curve that will have to be addressed if you go that route.

For those of us who know nothing of your project (beyond a quick glance at https://github.com/OpenFTC/OpenRC-Turbo/), can you give a little more specifics on what you would like to achieve? For example, it appears that the project has some code intended only to run on android, or only in a runtime where `com.qualcomm.*` jars are provided - these won't make sense to transpile for the browser, either with j2cl or with jsweet. Some projects also seem to come with cpp code, which also won't transpile (at least without another step, like clang to wasm) to run in a browser with these tools.

**Reply by jweiland1** [reply]
Hi sorry for any confusion.

We have a project (the VRS) where we want students to write Java code that then gets translated in the JavaScript so that there code can communicate with our Unity / webgl.  see https://www.vrobotsim.online/programpage.html    Our current approach was to take each Java program that was written and individually make it work in JavaScript sort of like creating a dictionary.   so our original transpiler is this Dictionary like project .

note;
OpenRc- is where the java libraries are-
We are not using Android 
We have a maven issue
see  https://github.com/jweiland1/VRS-Local-ElectronApp-base/tree/main-ug/assets/js 

Thanks
